src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -92,12 +92,11 @@ class SetupPyNativeTools(datatype([
   """
 
 
-@rule(SetupPyNativeTools, [Select(CCompiler), Select(CppCompiler), Select(Linker), Select(Platform)])
-def get_setup_py_native_tools(c_compiler, cpp_compiler, linker, platform):
+@rule(SetupPyNativeTools, [Select(LLVMCToolchain), Select(LLVMCppToolchain), Select(Platform)])
```

Comment:
This seems like a step back - `SetupPyNativeTools` which only cares about a `CppToolchain` and a `CToolchain` is not LLVM specific and cannot be produced for GCC. Am I missing something or is a TODO/comment missing?

Reply:
Totally right! In 3441e74 I moved all the rules and datatypes into `python_native_code.py`. The rules `select_c_toolchain_for_local_dist_compilation` and `select_cpp_toolchain_for_local_dist_compilation` mean that LLVM is used as the default toolchain, but `SetupPyNativeTools` only requires an injected `CToolchain` and `CppToolchain`.